src/features/emails/components/EmailDiagramHoverCard.tsx

```diff
@@ -0,0 +1,133 @@
+import { FC } from 'react';
+import { FormattedTime } from 'react-intl';
+import { Box, Divider, Paper, Typography } from '@mui/material';
+
+import { Msg } from 'core/i18n';
+import ZUIDuration from 'zui/ZUIDuration';
+import messageIds from '../l10n/messageIds';
+import { EmailInsights, ZetkinEmailStats } from '../types';
+import getRelevantDataPoints from '../utils/getRelevantDataPoints';
+import { ZetkinEmail } from 'utils/types/zetkin';
+
+type Props = {
+  mainInsights: EmailInsights;
+  mainStats: ZetkinEmailStats | null;
```

Comment:
You're right about all of these, and it might be possible to express stricter types as unions of the various states. I think it sounds great to explore it!

For the states that concern loading, I would actually suggest a bigger discussion about how we should be loading data. The `IFuture` path that we chose to explore (inspired by Dart/Flutter) has proven to have problems in React, and it might be better to completely shift to [`Suspense`](https://react.dev/reference/react/Suspense). This has been on the horizon for a while, which is why you're seeing a lot of "sloppiness" with loading states.